uuid.go

```diff
@@ -26,6 +37,15 @@ func GenerateUUID() (string, error) {
 	return FormatUUID(buf)
 }
 
+// GenerateUUIDByReader is used to generate a random UUID by a given Reader
```

Comment:
I didn't see this used in your patch, is that upcoming?

Reply:
yes